R/dist.R

```diff
@@ -1382,48 +1474,374 @@ plot.dist_spec <- function(x, ...) {
-      mean_sd = 0,
-      sd_sd = 0,
-      distribution = x$dist,
-      max = c(x$max)
-    )
-  } else if (strategy == "sample") {
-    lower_bound <- ifelse(x$dist == "gamma", 0, -Inf)
-    mean <- rtruncnorm(
-      n = 1, a = lower_bound, mean = x$mean_mean, sd = x$mean_sd
+
+  ret <- lapply(x, function(x) {
+    ## if x is fixed already we don't have to do anything
+    if (
+      x$distribution == "nonparametric" ||
+      all(vapply(x$parameters, is.numeric, logical(1)))
+    ) {
+      return(x)
+    }
+    ## apply strategy depending on choice
+    if (strategy == "mean") {
+      x$parameters <- lapply(x$parameters, mean)
+    } else if (strategy == "sample") {
+      lower_bound <-
+        lower_bounds(x$distribution)[natural_params(x$distribution)]
+      mean <- as.list(rtruncnorm(
+        n = 1, a = lower_bound,
+        mean = vapply(x$parameters, mean, numeric(1)),
+        sd = vapply(x$parameters, sd_dist, numeric(1))
+      ))
+      names(mean) <- names(x$parameters)
+      x$parameters <- mean
+    }
+    return(x)
+  })
+
+  attr(ret, "class") <- c("dist_spec", "list")
+  return(ret)
+}
+
+##' @details
+##' Probability distributions are ubiquitous in EpiNow2, usually representing
+##' epidemiological delays (e.g., the generation time for delays between
+##' becoming infecting and infecting others; or reporting delays)
+##'
+##' They are generated using functions that have a name corresponding to the
+##' probability distribution that is being used. They generated `dist_spec`
+##' objects that are then passed to the models underlying EpiNow2.
+##
+##' All parameters can be given either as fixed values (a numeric value) or as
+##' uncertain values (a `dist_sepc`). If given as uncertain values, currently
+##' only normally distributed parameters (generated using `Normal()`) are
+##' supported.
+##'
+##' Each distribution has a representation in terms of "natural" parameters (the
+##' ones used in stan) but can sometimes also be specified using other
+##' parameters such as the mean or standard deviation of the distribution. If
+##' not given as natural parameters then these will be calculated from the given
+##' parameters. If they have uncertainty, this will be done by random sampling
+##' from the given uncertainty and converting resulting parameters to their
+##' natural representation.
+##'
+##' Currently available distributions are lognormal, gamma, normal, fixed
+##' (delta) and nonparametric. The nonparametric is a special case where the
+##' probability mass function is given directly as a numeric vector.
+##'
+##' @inheritParams stats::Lognormal
+##' @param mean,sd mean and standard deviation of the distribution
+##' @param max Numeric, maximum value of the distribution. The distribution will
+##' be truncated at this value. Default: `Inf`, i.e. no maximum.
+##' @return A `dist_spec` representing a distribution of the given
+##'   specification.
+##' @author Sebastian Funk
+##' @export
+##' @rdname Distributions
```

Comment:
Issue with roxygen here and elsewhere being ##` rather than #`

Reply:
85ec4e2a